### ISSUE TITLE
Mongoose browser implementation

### ISSUE THREAD

**Issue by gilles-yvetot** [Automattic/mongoose#9273]
**Do you want to request a *feature* or report a *bug*?**
Feature -> docs

**Problem**

@vkarpov15 I am not sure to understand the mongoose browser documentation. I have been a user of mongoose for many years now (❤️ ) but I am struggling to understand it. 
- You said there is no pre-build library of mongoose browser but I am still seeing one when installing mongoose on my project
- I am using mongoose to define my schema and models in my library. This project is later consumed by all my backends projects. I'd love to be able to create a different build from the same schemas I have defined. But does it mean I need to re-write all of them and replacing `const mongoose = require('mongoose')` with `const mongoose = require('mongoose/browser')`? I am using ES6 everywhere

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node 12
Mongoose 5.9.25 
Mongo 4.2

**Comment by vkarpov15** [comment]
As long as you use Webpack and your [webpack config](https://masteringjs.io/tutorials/webpack/config) has `target: 'web'`, then you should be able to use `const mongoose = require('mongoose')` in your browser code.

Thanks for pointing out that we already have a pre-built bundle, this section of the docs is out of date after #8435

**Comment by vkarpov15** [comment]
I found a bit of a bug in the UMD build: the below script would fail because older versions of Babel didn't support `extends Error` very well. I upgraded Babel and can confirm the below script works with a [basic webpack config](https://masteringjs.io/tutorials/webpack/config).

```javascript
import mongoose from 'mongoose';

    // Mongoose's browser library does **not** have models. It only supports
    // schemas and documents. The primary use case is validating documents
    // against Mongoose schemas.
    const doc = new mongoose.Document({}, new mongoose.Schema({
      name: { type: String, required: true }
    }));
    // Prints an error because `name` is required.
    console.log(doc.validateSync());
```

**Comment by vkarpov15** [comment]
Just to clarify, you can still use `import mongoose from 'mongoose/browser';`, but then you're using your own Webpack config to compile Mongoose.

**Comment by gilles-yvetot** [comment]
@vkarpov15 I don't know if I should open a new ticket or not. 
I have tried the version 5.10.3 and everything is working fine with the recommended webpack config. I do have a warning for `mongodb-client-encryption` not being found

```
Module not found: Error: Can't resolve 'mongodb-client-encryption' in '/Users/gilles/Documents/omni/app/packages/extension/node_modules/mongodb/lib/operations'
 @ ./node_modules/mongodb/lib/operations/connect.js
 @ ./node_modules/mongodb/lib/mongo_client.js
 @ ./node_modules/mongodb/index.js
 @ ./node_modules/mongoose/lib/index.js
 @ ./node_modules/mongoose/index.js
```

**Comment by vkarpov15** [comment]
@gilles-yvetot are you using `import mongoose from 'mongoose/browser'` or `import mongoose from 'mongoose'`?

You should be safe to ignore that warning or stub out mongodb-client-encryption using [webpack externals](https://masteringjs.io/tutorials/webpack/externals).
